File: ADSO/Lenguajes/Ruby/ejem_while_funt.rb
def contador
  j = 0

  puts '--------------------------------------'
  puts 'Para detener el programa escribe "stop"'
  puts '--------------------------------------'

  while true
    print "Introducir numero: "
    n = gets.chomp

    if n.downcase == "stop"
      break
    else
      m = n.to_i
      j += m
    end
  end

  puts '-------------------------'
  puts j
end

puts "Ejecutar programa Y/N"
y_n = gets.chomp.downcase

if y_n == "y"
  contador
end

File: ADSO/Lenguajes/Dart/ejem_while_funt.dart
import 'dart:io';

void contador(){
  int j = 0;

  print('--------------------------------------');
  print('Para detener el programa escribe "stop"');
  print('--------------------------------------');

  while (true){

    print("Introducir numero:");
    String? n = stdin.readLineSync();

    n = n?.toLowerCase();

    if (n == "stop" ) {
      break;
    }else {
      int m = int.parse(n!);
      j = j + m;
    }
  }
  print("-------------------------");

  print(j);
}

void main(){
  
  print("Ejecutar programa Y/N");
  String? y_n = stdin.readLineSync();
  y_n = y_n?.toLowerCase();

  if (y_n == "y") {
    contador();
  }
}